homeassistant/components/climate/systemair_savecair.py

```diff
@@ -0,0 +1,286 @@
+"""
+Support for Systemair Savecair climate devices.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/climate.systemair_savecair/
+"""
+import voluptuous as vol
+
+from homeassistant.components.climate import (
+    PLATFORM_SCHEMA, STATE_AUTO, STATE_IDLE, STATE_MANUAL, SUPPORT_AWAY_MODE,
+    SUPPORT_FAN_MODE, SUPPORT_ON_OFF, SUPPORT_OPERATION_MODE,
+    SUPPORT_TARGET_TEMPERATURE, ClimateDevice)
+from homeassistant.const import (
+    ATTR_TEMPERATURE, CONF_NAME, CONF_PASSWORD, TEMP_CELSIUS, TEMP_FAHRENHEIT)
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['python-systemair-savecair==0.0.6']
+
+ATTR_SUPPLY_AIR_TEMPERATURE = "supply_air_temperature"
+ATTR_SUPPLY_AIR_FAN_SPEED = "supply_air_fan_speed"
+ATTR_EXTRACT_AIR_FAN_SPEED = "extract_air_fan_speed"
+
+FAN_OFF = "Off"
+FAN_LOW = "Low"
+FAN_NORMAL = "Medium"
+FAN_HIGH = "High"
+
+FAN_LIST = [FAN_OFF, FAN_LOW, FAN_NORMAL, FAN_HIGH]
+
+OPERATION_LIST = [STATE_AUTO, STATE_MANUAL]
+
+SUPPORT_FLAGS = SUPPORT_TARGET_TEMPERATURE | \
+                SUPPORT_FAN_MODE | \
+                SUPPORT_OPERATION_MODE | \
+                SUPPORT_AWAY_MODE | \
+                SUPPORT_ON_OFF
+
+SAVECAIR_FAN_MODES = {
+    1: FAN_OFF,
+    2: FAN_LOW,
+    3: FAN_NORMAL,
+    4: FAN_HIGH
+}
+
+SAVECAIR_OPERATION_MODES = {
+    0: STATE_AUTO,
+    1: STATE_IDLE,
+    2: STATE_MANUAL,
+}
+
+HA_OPERATION_MODES = {
+    val: key for key, val in SAVECAIR_OPERATION_MODES.items()
+}
+
+
+CONF_IAM_ID = "iam_id"
+DEFAULT_NAME = "SystemAIR"
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Required(CONF_IAM_ID): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string
+})
+
+
+async def async_setup_platform(hass, config, async_add_entities,
+                               discovery_info=None):
+    """Set up the SystemAIRplatform."""
+    name = config[CONF_NAME]
+    iam_id = config[CONF_IAM_ID]
+    password = config[CONF_PASSWORD]
+
+    # Create savecair client
+    from systemair.savecair import SaveCairClient
+    client = SaveCairClient(iam_id, password, loop=hass.loop)
```

Comment:
We should check that the credentials are correct before continuing with set up. If they are bad, log error and return.